services/galley/src/Galley/API/MLS/Message.hs

```diff
@@ -843,11 +863,9 @@ processInternalCommit qusr senderClient con lconv mlsMeta cm epoch action sender
                 qusr
                 (Set.singleton (creatorClient, creatorRef))
             (Left _, SelfConv, _) ->
-              throw . InternalErrorWithDescription $
-                "Unexpected creator client set in a self-conversation"
-            -- this is a newly created conversation, and it should contain exactly one
-            -- client (the creator)
-
+              -- this is a newly created conversation, and it should contain exactly one
+              -- client (the creator)
+              throwS @'MLSUnexpectedSenderClient
```

Comment:
This updates error retporting of the self conversation, not the GTC. Can you please update PR description / title?